src/mavsdk/core/callback_list_impl.h

```diff
@@ -16,15 +17,25 @@ template<typename... Args> class CallbackListImpl {
     Handle<Args...> subscribe(const std::function<void(Args...)>& callback)
     {
         check_removals();
+        process_subscriptions();
 
         // We need to return a handle, even if the callback is nullptr to
         // unsubscribe. That's fine, the handle just won't remove anything
         // when/if used later.
-        auto handle = Handle<Args...>(_last_id++);
+        auto handleId = _last_id.fetch_add(1, std::memory_order_relaxed);
```

Comment:
How did this work before without an atomic id?

Reply:
I think the issue only happens if you add a ton of subscriptions from different threads simultaneously - kind of a rare scenario, so we didn’t see it earlier.